string-constants-doc/string-constants/translation-guide.scrbl

```diff
@@ -0,0 +1,161 @@
+the file
+@link["https://github.com/racket/string-constants/blob/master/string-constants-lib/string-constants/private/english-string-constants.rkt"]{@tt{english-string-constants.rkt}}
+to @tt{yourlanguage-string-constants.rkt}.
+
+Basically English is the master file containing the originals of the
+strings.  Therefore, start with a copy of that.  Note that if you
+continue to update and maintain your translation — English will be
+the file you will sync with.
+
+Once you have copied the file you will need to declare it so DrRacket
+knows it exists and has to be included with the rest of the
+translations in the official builds.  To do that just edit the file
+@link["https://github.com/racket/string-constants/blob/master/string-constants-lib/string-constants/string-constant.rkt"]{@tt{string-constants/string-constants-lib/string-constants/string-constant.rkt}}
+and add your language in two places:
+@racketblock[
+(require (prefix-in english: "private/english-string-constants.rkt")
+…
+          (prefix-in yourlanguage: "private/yourlanguage-string-constants.rkt"))
+]
+
+and
+@racketblock[
+(define built-in-string-constant-sets
+  (list
+   (make-sc 'english english:string-constants #f)
+…
+   (make-sc ' yourlanguage yourlanguage:string-constants #f)))
+]
+
+And with that you are done with the initial work — you have created
+a template for your language (still filled in with the English
+originals) and you have told Racket to use it. We are off to the races
+and all we need to do is fill in the translations.
+
+
+@; ----------------------------------------------------------------------
+
+@section{Translation}
+
+Open @tt{yourlanguage-string-constants.rkt} in your favorite text
+editor. It has a fairly trivial structure: it is basically a map — a
+long list of key-value pairs. The key is the identifier of the string
+— how Racket source code is going to refer to the message. Do not
+change these. Value is the translation. As you started with the
+English version — you have all the comments and the formatting. I
+highly recommend keeping these in place because they are very helpful
+when you need to update the translations.
+
+The first thing you need to do is fix the line:
+@racketblock[(module english-string-constants "string-constant-lang.rkt"
+…)]
+
+It has to become
+@racketblock[(module yourlanguage-string-constants "string-constant-lang.rkt"
+…)]
+
+
+The second thing is to add some metadata to your file. You can use
+standard comments in the file – either prefix a line with double
+semicolon @tt{;;} or comment out a block of text via @tt{#| commented
+block |#}.
+
+In that part of the file put some basic data — like your name, which
+version of the @tt{private/english-string-constants.rkt} file you are
+syncing with. This is also the place to put out conventions for the
+translation such as specific terms, style etc. I would also recommend
+spelling out the license of it. For example in the
```

Comment:
Maybe "spelling out the license you plan to use". For anything that is to be distributed with Racket, the maintainers will be more likely to accept it if you use the strategy below, so probably a good idea to say that.

Reply:
Done but double check

Reply:
Looks great.